Gems/Terrain/Code/Source/Components/TerrainWorldRendererComponent.cpp

```diff
@@ -93,20 +100,77 @@ namespace Terrain
                     ;
 
                 editContext->Class<ClipmapConfiguration>("Clipmap", "Settings related to clipmap rendering")
+                    ->DataElement(AZ::Edit::UIHandlers::ComboBox, &ClipmapConfiguration::m_clipmapSize,
+                        "Clipmap image size",
+                        "The size of the clipmap image in each layer.")
+                        ->EnumAttribute(ClipmapConfiguration::ClipmapSize2048, "2048")
+                        ->EnumAttribute(ClipmapConfiguration::ClipmapSize1024, "1024")
+                        ->EnumAttribute(ClipmapConfiguration::ClipmapSize512, "512")
+                    ->DataElement(AZ::Edit::UIHandlers::Slider, &ClipmapConfiguration::m_macroClipmapMaxRenderRadius,
+                        "Macro clipmap max render radius: m",
+                        "Max render radius that the lowest resolution clipmap can cover. \n"
+                        "For example, 1000 means the clipmaps render 1000 meters at most from the view position.")
+                        ->Attribute(AZ::Edit::Attributes::Step, 1.0f)
+                        ->Attribute(AZ::Edit::Attributes::Min, 0.0f)
+                        ->Attribute(AZ::Edit::Attributes::SoftMin, 1000.0f)
+                        ->Attribute(AZ::Edit::Attributes::SoftMax, 10000.0f)
+                        ->Attribute(AZ::Edit::Attributes::Max, 400000.0f)
+                    ->DataElement(AZ::Edit::UIHandlers::Slider, &ClipmapConfiguration::m_detailClipmapMaxRenderRadius,
+                        "Detail clipmap max render radius: m",
+                        "Max render radius that the lowest resolution clipmap can cover. \n"
+                        "For example, 1000 means the clipmaps render 1000 meters at most from the view position.")
+                        ->Attribute(AZ::Edit::Attributes::Step, 1.0f)
+                        ->Attribute(AZ::Edit::Attributes::Min, 0.0f)
+                        ->Attribute(AZ::Edit::Attributes::SoftMin, 100.0f)
+                        ->Attribute(AZ::Edit::Attributes::SoftMax, 2000.0f)
+                        ->Attribute(AZ::Edit::Attributes::Max, 10000.0f)
                     ->DataElement(AZ::Edit::UIHandlers::Slider, &ClipmapConfiguration::m_macroClipmapMaxResolution,
-                        "Macro clipmap max resolution (in texels per meter)",
+                        "Macro clipmap max resolution: texels/m",
                         "The resolution of the highest resolution clipmap in the stack.")
-                        ->Attribute(AZ::Edit::Attributes::Min, 0.5f)
+                        ->Attribute(AZ::Edit::Attributes::Min, 0.0f)
```

Comment:
What happens if you set this to 0? Does it disable macro clipmap drawing?

Reply:
Increased this to 0.1 since 0.0 will break. It should be intended to break if it approaches the hard limit.